server/api/graphql_loader_run.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"context"
+
+	"github.com/graph-gophers/dataloader/v7"
+	"github.com/mattermost/mattermost-plugin-playbooks/server/app"
+)
+
+func graphQLStatusPostsLoader[V []app.StatusPost](ctx context.Context, keys []string) []*dataloader.Result[V] {
+	result := make([]*dataloader.Result[V], len(keys))
+	if len(keys) == 0 {
+		return result
+	}
+
+	c, err := getContext(ctx)
+	if err != nil {
+		return populateResultWithError(err, result)
+	}
+
+	statusPostsByRunID, err := c.runStore.GetStatusPostsByIDs(keys)
+	if err != nil {
+		return populateResultWithError(err, result)
+	}
+
+	for i, runID := range keys {
+		statusPosts, ok := statusPostsByRunID[runID]
+		if !ok {
+			result[i] = &dataloader.Result[V]{Data: nil}
+			continue
+		}
+		result[i] = &dataloader.Result[V]{
+			Data: V(statusPosts),
+		}
+	}
+
+	return result
+}
+
+func graphQLTimelineEventsLoader[V []app.TimelineEvent](ctx context.Context, keys []string) []*dataloader.Result[V] {
+	result := make([]*dataloader.Result[V], len(keys))
+	if len(keys) == 0 {
+		return result
+	}
+
+	c, err := getContext(ctx)
+	if err != nil {
+		return populateResultWithError(err, result)
+	}
+
+	timelineEvents, err := c.runStore.GetTimelineEventsByIDs(keys)
+	if err != nil {
+		return populateResultWithError(err, result)
+	}
+
+	timelineEventsByRunID := make(map[string][]app.TimelineEvent)
```

Comment:
nit: would it be consistent to use the generic type here?
```suggestion
	timelineEventsByRunID := make(map[string]V)
```